sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs

```diff
@@ -214,35 +250,31 @@ public virtual async Task<Response<TableBatchResponse>> SubmitBatchAsync(Cancell
         }
 
         /// <summary>
-        /// Submits the batch transaction to the service for execution.
-        /// The sub-operations contained in the batch will either succeed or fail together as a transaction.
+        /// Tries to get the entity that caused the batch operation failure from the <see cref="RequestFailedException"/>.
         /// </summary>
-        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns><see cref="Response{T}"/> containing a <see cref="TableBatchResponse"/>.</returns>
-        /// <exception cref="RequestFailedException"/> if the batch transaction fails./>
-        public virtual Response<TableBatchResponse> SubmitBatch(CancellationToken cancellationToken = default)
+        /// <param name="exception">The exception thrown from <see cref="TableTransactionalBatch.SubmitBatch(CancellationToken)"/> or <see cref="TableTransactionalBatch.SubmitBatchAsync(CancellationToken)"/>.</param>
+        /// <param name="failedEntity">If the return value is <c>true</c>, contains the <see cref="ITableEntity"/> that caused the batch operation to fail.</param>
+        /// <returns><c>true</c> if the failed entity was retrieved from the exception, else <c>false</c>.</returns>
+        public bool TryGetFailedEntityFromException(RequestFailedException exception, out ITableEntity failedEntity)
         {
-            var messageList = BuildOrderedBatchRequests();
+            failedEntity = null;
 
-            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(SubmitBatch)}");
-            scope.Start();
-            try
+            if (exception.Data.Contains(TableConstants.ExceptionData.FailedEntityIndex))
             {
-                var request = _tableOperations.CreateBatchRequest(_batch, null, null);
-                var response = _tableOperations.SendBatchRequest(request, messageList, cancellationToken);
-
-                for (int i = 0; i < response.Value.Count; i++)
+                try
                 {
-                    messageList[i].HttpMessage.Response = response.Value[i];
+                    if (int.TryParse(exception.Data[TableConstants.ExceptionData.FailedEntityIndex] as string, out int index))
```

Comment:
Why is it a string?

Reply:
It complained when I tried to make it an `int` since it's not a reference type. I suppose I could make it `int?`